### PR TITLE
Fixes: #158 - Add case insensitive sort options.

### PR DESCRIPTION
**Pull request by fiveNinePlusR**
The two options for sorting:

* `alphabetic-case-insensitive-asc`
* `alphabetic-case-insensitive-desc`

### CODE REVIEW

**Comment by Alexander-Miller** [comment]
> All checks have failed

Do you happen to know your way around travis? I'm pretty sure my emacs26 builds are allowed to fail, but the setting is not applied.

File: src/elisp/treemacs-branch-creation.el [hunk 1]
```diff
@@ -43,13 +43,21 @@ is a marker pointing to POS."
   (copy-marker pos t))
 
 (defsubst treemacs--sort-alphabetic-asc (f1 f2)
-  "Sort F1 and F2 alphabetically asc."
+  "Sort f1 and f2 alphabetically asc."
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-branch-creation.el]: Function params must be capitalized or the compiler complains. `make test` can catch these (this stuff will go into a Contributing readme section).

File: src/elisp/treemacs-branch-creation.el [hunk 2]
```diff
@@ -99,6 +107,8 @@ DEFAULT: Face"
           (-pcase treemacs-sorting
             [`alphabetic-asc  #'treemacs--sort-alphabetic-asc]
             [`alphabetic-desc #'treemacs--sort-alphabetic-desc]
+            [`alphabetic-case-insensitive-asc  #'treemacs--sort-alphabetic-case-insensitive-asc]
+            [`alphabetic-case-insensitive-desc #'treemacs--sort-alphabetic-case-insensitive-desc]
             [`size-asc        #'treemacs--sort-size-asc]
```
Comment (Alexander-Miller) [inline src/elisp/treemacs-branch-creation.el]: You should also delete the extra whitespace in the other match arms. I try to keep things nicely aligned, but not if the difference is as large as with the new values.

**Comment by fiveNinePlusR** [comment]
I updated my branch to allow failures in emacs 26

**Comment by fiveNinePlusR** [comment]
Is a rebase and squash something that you'd like to see happen to clean up the history?

**Comment by Alexander-Miller** [comment]
Yes, so long as the changes are small enough there should be one commit per topic, so the indent update should be squashed.

And thanks for setting travis straight.

**Comment by fiveNinePlusR** [comment]
I am not sure how to mark that the changes have been made to the commits inside github but it's been updated.

**Comment by Alexander-Miller** [comment]
I don't think github has that feature. At any rate we're good to merge :+1:

**Pull request merged**